modules/tensor_mechanics/src/actions/LineElementAction.C

```diff
@@ -0,0 +1,834 @@
+      paramError("density",
+                 "LineElementAction: Either name of the density material property or a constant "
+                 "density value should be provided as input using `density` for creating the "
+                 "consistent mass/inertia matrix required for dynamic beam simulation.");
+
+    if (_dynamic_nodal_translational_inertia && !isParamValid("nodal_mass"))
+      paramError("nodal_mass",
+                 "LineElementAction: `nodal_mass` should be provided as input to calculate "
+                 "inertial forces on beam due to nodal mass.");
+
+    if (_dynamic_nodal_rotational_inertia &&
+        ((!isParamValid("nodal_Ixx") || !isParamValid("nodal_Iyy") || !isParamValid("nodal_Izz"))))
+      mooseError("LineElementAction: `nodal_Ixx`, `nodal_Iyy`, `nodal_Izz` should be provided as "
+                 "input to calculate inertial torque on beam due to nodal inertia.");
+  }
+  else // if truss
+  {
+    if (!isParamValid("youngs_modulus") || !isParamValid("area"))
+      mooseError("LineElementAction: `youngs_modulus` and `area` should be provided as input for "
+                 "truss elements.");
+
+    if (isParamValid("rotations"))
+      paramError("rotations",
+                 "LineElementAction: Rotational variables cannot be set for truss elements.");
+  }
+}
+
+void
+LineElementAction::act()
+{
+  // Get the subdomain involved in the action once the mesh setup is complete
+  if (_current_task == "setup_mesh_complete")
+  {
+    // get subdomain IDs
+    for (auto & name : _subdomain_names)
+      _subdomain_ids.insert(_mesh->getSubdomainID(name));
+  }
+
+  //
+  // Gather info from all other LineElementAction
+  //
+  actGatherActionParameters();
+
+  //
+  // Add variables (optional)
+  //
+  actAddVariables();
+
+  //
+  // Add Materials - ComputeElasticityBeam, ComputeIncrementalBeamStrain or ComputeFiniteBeamStrain
+  // and ComputeBeamResultants for beam elements and LinearElasticTruss for truss elements
+  //
+  actAddMaterials();
+
+  //
+  // Add Kernels - StressDivergenceBeam and InertialForceBeam (if dynamic_consistent_inertia is
+  // turned on) for beams and StressDivergenceTensorsTruss for truss elements
+  //
+  actAddKernels();
+
+  //
+  // Add aux variables for translational and Rotational velocities and acceleration for dynamic
+  // analysis using beams
+  //
+  actAddAuxVariables();
+
+  //
+  // Add NewmarkVelAux and NewarkAccelAux auxkernels for dynamic simulation using beams
+  //
+  actAddAuxKernels();
+
+  //
+  // Add NodalKernels - NodalTranslationalInertia (if dynamic_nodal_translational_inertia is turned
+  // on) and NodalRotattionalInertia (if dynamic_nodal_rotational_inertia) for dynamic simulations
+  // using beams
+  //
+  actAddNodalKernels();
+}
+
+void
+LineElementAction::actGatherActionParameters()
+{
+  //
+  // Gather info about all other master actions when we add variables
+  //
+  if (_current_task == "add_variable" && getParam<bool>("add_variables"))
```

Comment:
It probably doesn't make too much difference, but I think I would have put these checks for `_current_task` in the `act()` method that calls these.